src/models/state/mod.rs

```diff
@@ -1472,19 +1460,21 @@ impl GlobalState {
                 )
                 .await?;
 
-            // Update mempool with UTXOs from this block. This is done by removing all transaction
-            // that became invalid/was mined by this block.
-            myself
+            // Update mempool with UTXOs from this block. This is done by
+            // removing all transaction that became invalid/was mined by this
+            // block.
+            let (mempool_events, update_jobs) = myself
                 .mempool
                 .update_with_block_and_predecessor(
                     &new_block,
                     &tip_parent,
-                    vm_job_queue,
-                    TritonVmJobPriority::Highest,
+                    myself.net.tx_proving_capability,
                     myself.cli().compose,
                 )
                 .await;
 
+            // TODO: Inform wallet about mempool_events
```

Comment:
If the idea is that we should not inform wallet until the proof and mutatorsetupdate has finished (which seems logical), then it seems that these methods should also return Ok((mempool_events, update_jobs)) and the caller can inform the wallet after the update jobs have completed successfully.

Of course that is gross as it separates the logic/location of where "inform wallet" actions take place into different modules.

A bigger problem with it is that the user will not see the spend reflected in their wallet balance or history until the jobs finish.  Which can be very disconcerting.  That is the case now of course, but we may have an opportunity to improve it, and should.

perhaps GlobalState::update_with_block_and_predecessor can inform the wallet of the mempool events as before, but the wallet could somehow detect / inform that these tx are incomplete/partial.

Failing something like that, we need to somehow inform the user that the send succeeded, but is expected to take minutes before it will be reflected in wallet balance/history.  I suppose that could just be client UI messaging.

something to think about.